### PR TITLE
Fix flaky tests

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Closes: #646
Refs: #647

* Fixes `SchemaValidationTest` which started failing because of an error message change in a newer version of `jsonschema` library
* Also fixes unhandled project rejection in `RepairingTask` which led to #647

See http://jenkins.hazelcast.com/job/NodeJS-10-master/322/testReport/(root)/invalid%20configuration%20is%20caught%20by%20the%20validator/SchemaValidationTest_invalid_configuration_is_caught_by_the_validator/ as a failure example

### CODE REVIEW

File: test/config/SchemaValidationTest.js [hunk 1]
```diff
@@ -40,6 +40,7 @@ describe('SchemaValidationTest', function () {
 
     it('invalid configuration is caught by the validator', function () {
         const invalidJson = fs.readFileSync(path.resolve(__dirname, 'configurations/invalid.json'), 'utf8');
-        expect(validateCandidate(invalidJson).errors[0]).to.exist.with.property('message', 'must have a minimum value of 1000');
+        expect(validateCandidate(invalidJson).errors[0]).to.exist.with.property('message');
```
Comment (mdumandag) [inline test/config/SchemaValidationTest.js:43]: Not a great deal but we may do the `validateCandidate` call once, store it in some variable and do the assertions

**Review by mdumandag — APPROVED** [review]

File: test/config/SchemaValidationTest.js (hunk 1 above)
Comment (puzpuzpuz) [reply test/config/SchemaValidationTest.js:43]: Yes, it makes sense. Let me fix it.

Comment (puzpuzpuz) [reply test/config/SchemaValidationTest.js]: Done.

**Comment by puzpuzpuz** [comment]
Test failure is unrelated with the PR. Created #646 to track it.

**Comment by puzpuzpuz** [comment]
verify

**Comment by puzpuzpuz** [comment]
@mdumandag could you take a look at e26ae8a486bc784e7c8522b865920193537525f6? It fixes a bug which led to flakiness in `MigratedDataTest`

**Pull request merged**